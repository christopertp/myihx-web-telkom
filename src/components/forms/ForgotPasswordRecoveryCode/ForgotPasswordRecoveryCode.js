import React from 'react';
import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';
import { Field } from 'redux-form';
import Button from '../../elements/Button';
import Spinner from '../../elements/Spinner';
import PasswordField from '../../fields/Password';
import styles from './styles.scoped.css';

export default function ForgotPasswordRecoveryCode(props) {
  const { handleSubmit, invalid } = props;
  const { isLoading } = useSelector(s => s.forgotPassword);
  const inputProps = {
    label: 'RECOVERY CODE',
    placeholder: 'Enter recovery code',
  };

  return (
    <form className={styles.root} onSubmit={handleSubmit}>
      <Field component={PasswordField} inputProps={inputProps} name="code" />
      <Button disabled={invalid || isLoading} type="submit">
        {isLoading ? <Spinner /> : 'Next'}
      </Button>
    </form>
  );
}

ForgotPasswordRecoveryCode.defaultProps = {
  handleSubmit: () => {},
  invalid: false,
};

ForgotPasswordRecoveryCode.propTypes = {
  handleSubmit: PropTypes.func,
  invalid: PropTypes.bool,
};
