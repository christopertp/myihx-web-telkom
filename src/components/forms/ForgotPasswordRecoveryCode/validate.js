export default function validate(values) {
  const { code } = values;

  return {
    code: !code ? 'Harus diisi' : '',
  };
}
