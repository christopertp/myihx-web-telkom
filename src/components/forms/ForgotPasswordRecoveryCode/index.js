import { reduxForm } from 'redux-form';
import ForgotPasswordRecoveryCode from './ForgotPasswordRecoveryCode';
import validate from './validate';

export default reduxForm({
  initialValues: {
    code: '',
  },
  form: 'forgotPasswordRecoveryCode',
  validate,
})(ForgotPasswordRecoveryCode);
