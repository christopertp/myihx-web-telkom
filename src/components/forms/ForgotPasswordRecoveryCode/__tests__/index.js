import React from 'react';
import { useSelector } from 'react-redux';
import ShallowRenderer from 'react-test-renderer/shallow';
import Spinner from '../../../elements/Spinner';
import ForgotPasswordRecoveryCode from '../ForgotPasswordRecoveryCode';

useSelector.mockImplementation(fn => {
  fn(({ login: '' }));
  return { isLoading: false, message: '' };
});

describe('src/components/forms/ForgotPasswordRecoveryCode', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<ForgotPasswordRecoveryCode />);
    expect(tree).toMatchSnapshot();

    useSelector.mockImplementationOnce(() => ({ isLoading: true, message: 'error' }));
    ForgotPasswordRecoveryCode.defaultProps.handleSubmit();
    const result = ForgotPasswordRecoveryCode({
      ...ForgotPasswordRecoveryCode.defaultProps,
    });
    const button = result.props.children[1];
    expect(button.props.children.type).toBe(Spinner);
  });
});
