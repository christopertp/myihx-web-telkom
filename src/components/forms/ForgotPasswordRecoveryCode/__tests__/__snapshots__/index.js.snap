// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/forms/ForgotPasswordRecoveryCode render 1`] = `
<form
  onSubmit={[Function]}
>
  <Field
    component={[Function]}
    inputProps={
      Object {
        "label": "RECOVERY CODE",
        "placeholder": "Enter recovery code",
      }
    }
    name="code"
  />
  <Button
    className=""
    disabled={false}
    onClick={null}
    to=""
    type="submit"
    variant="red"
  >
    Next
  </Button>
</form>
`;
